avm/res/container-instance/container-group/main.bicep

```diff
@@ -146,52 +129,85 @@ resource containergroup 'Microsoft.ContainerInstance/containerGroups@2023-05-01'
+    // so the only option is the json() function for memoryInGB,
+    // and it means to copy everything else prop by prop
+    containers: [
+      for (container, index) in containers: {
+        name: container.name
+        properties: {
+          command: container.properties.?command
+          environmentVariables: container.properties.?environmentVariables
+          image: container.properties.?image
+          livenessProbe: container.properties.?livenessProbe
+          ports: container.properties.?ports
+          readinessProbe: container.properties.?readinessProbe
+          resources: {
+            requests: {
+              cpu: container.properties.resources.requests.cpu
+              gpu: container.properties.resources.requests.?gpu
+              memoryInGB: json(container.properties.resources.requests.?memoryInGB)
+            }
+            limits: !empty(container.properties.resources.?limits)
+              ? {
+                  cpu: container.properties.resources.?limits.?cpu
+                  gpu: container.properties.resources.?limits.?gpu
+                  memoryInGB: !empty(container.properties.resources.?limits.?memoryInGB)
+                    ? json(container.properties.resources.?limits.?memoryInGB!)
+                    : null
+                }
               : null
-            keyName: customerManagedKey!.keyName
-            keyVersion: !empty(customerManagedKey.?keyVersion ?? '')
-              ? customerManagedKey!.keyVersion
-              : (customerManagedKey.?autoRotationEnabled ?? true)
-                  ? null
-                  : last(split(cMKKeyVault::cMKKey.properties.keyUriWithVersion, '/'))
-            vaultBaseUrl: cMKKeyVault.properties.vaultUri
           }
-        : null
-      imageRegistryCredentials: imageRegistryCredentials
-      initContainers: initContainers
-      restartPolicy: restartPolicy
-      osType: osType
-      ipAddress: {
-        type: ipAddressType
-        autoGeneratedDomainNameLabelScope: !empty(dnsNameServers) ? autoGeneratedDomainNameLabelScope : null
-        dnsNameLabel: dnsNameLabel
-        ports: ipAddressPorts
+          securityContext: container.properties.?securityContext
+          volumeMounts: container.properties.?volumeMounts
+        }
       }
-      sku: sku
-      subnetIds: !empty(subnetResourceId)
-        ? [
-            {
-              id: subnetResourceId
-            }
-          ]
-        : null
-      volumes: volumes
-    },
-    !empty(dnsNameServers)
+    ]
+    encryptionProperties: !empty(customerManagedKey)
       ? {
-          dnsConfig: {
-            nameServers: dnsNameServers
-            searchDomains: dnsSearchDomains
-          }
+          identity: !empty(customerManagedKey.?userAssignedIdentityResourceId) ? cMKUserAssignedIdentity.id : null
+          vaultBaseUrl: cMKKeyVault.properties.vaultUri
+          keyName: customerManagedKey!.keyName
+          // TODO: keyRotation support if possible
```

Comment:
Is it supported?

Reply:
IDK. There is a [keyVersion](https://learn.microsoft.com/en-us/azure/templates/microsoft.containerinstance/containergroups?pivots=deployment-language-bicep#encryptionproperties) param, which is required, but no word on how to configure that with key rotation.

Reply:
I believe I worked on those of these CMK implementations. Not too long ago we introduced the auto-rotation where possible (alongside the AVM interface for it) and when doing that I recall testing exactly what you describe across all modules to find out which seem to support auto-rotation, and which don't.
It would appear this is one of those where it is not possible and one has to provide a version to the module / have the module pick the latest upon deployment.

Given the comment I figured you may know something I don't and there's another way 😄

Reply:
OK, I have rephrased the comment, it's just a reminder for later if the key rotation might be supported